spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java

```diff
@@ -813,11 +814,11 @@ public void rollbackOnExceptionIfNecessary(Throwable ex) {
 	/**
 	 * Perform a commit or message acknowledgement, as appropriate.
 	 * @param localTx Whether the channel is locally transacted.
+	 * @param messageAckListener MessageAckListener set on the message listener.
 	 * @return true if at least one delivery tag exists.
 	 * @throws IOException Any IOException.
 	 */
-	public boolean commitIfNecessary(boolean localTx) throws IOException {
-
+	public boolean commitIfNecessary(boolean localTx, MessageAckListener messageAckListener) throws IOException {
```

Comment:
This is a public API; we can't change it in a point release; instead, inject the listener after creating the consumer.